include/seqan3/core/metafunction/pack.hpp

```diff
@@ -0,0 +1,44 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2018, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2018, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Hannes Hauswedell <hannes.hauswedell AT fu-berlin.de>
+ * \brief Provides various metafunctions on a set of types, usually provided as template argument pack.
+ */
+
+#pragma once
+
+#include <type_traits>
+
+#include <seqan3/core/platform.hpp>
+
+namespace seqan3::detail
+{
+
+//!\addtogroup metafunction
+//!\{
+
+//!\brief A type trait that indicates whether the first template argument is contained in the remaining.
+template<typename target_t, typename ...pack>
+struct type_in_pack : std::false_type {};
```

Comment:
this file should be merged/unified with the proposal #570 

Maybe `parameter_pack_contains` or something; It should match the naming scheme which will be decided in  #570!